src/platform/windows.rs

```diff
@@ -171,3 +173,30 @@ impl DeviceIdExtWindows for DeviceId {
         self.0.persistent_identifier()
     }
 }
+
+/// Additional methods on `Icon` that are specific to Windows.
+pub trait IconExtWindows: Sized {
+    /// Create an icon from a file path.
+    ///
+    /// Specify `size` to load a specific size from the file, or `None` to load the default
+    /// icon size from the file.
+    fn from_path<P: AsRef<Path>>(path: P, size: Option<(usize, usize)>) -> Result<Self, BadIcon>;
+
+    /// Create an icon from a resource embedded in this executable or library.
+    ///
+    /// Specify `size` to load a specific size from the file, or `None` to load the default
+    /// icon size from the file.
+    fn from_resource(ordinal: WORD, size: Option<(usize, usize)>) -> Result<Self, BadIcon>;
```

Comment:
`from_path` ditto. Also, the parameter name `ordinal` is unclear - `resource_id` or something similar would be better.